edge/pkg/clusterd/helper/helper.go

```diff
@@ -0,0 +1,149 @@
+/*
+Copyright 2015 The Kubernetes Authors.
+Licensed under the Apache License, Version 2.0 (the "License");
+you may not use this file except in compliance with the License.
+You may obtain a copy of the License at
+    http://www.apache.org/licenses/LICENSE-2.0
+Unless required by applicable law or agreed to in writing, software
+distributed under the License is distributed on an "AS IS" BASIS,
+WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+See the License for the specific language governing permissions and
+limitations under the License.
+
+*/
+
+package helper
+
+import (
+	"encoding/json"
+	"fmt"
+	"reflect"
+	"strings"
+
+	"k8s.io/klog/v2"
+
+	edgeclustersv1 "github.com/kubeedge/kubeedge/cloud/pkg/apis/edgeclusters/v1"
+	"github.com/kubeedge/kubeedge/edge/pkg/clusterd/config"
+	"github.com/kubeedge/kubeedge/edge/pkg/clusterd/util"
+)
+
+func GetLocalClusterScopeResourceNames(resType string, label string) []string {
+	labelOption := ""
+	if len(label) > 0 {
+		labelOption = "-l " + label
+	}
+	getResourceCmd := fmt.Sprintf(" %s get %s -o json %s --kubeconfig=%s | jq -r '.items[] | [.metadata.name] | @tsv' ", config.Config.KubectlCli, resType, labelOption, config.Config.Kubeconfig)
+	output, err := util.ExecCommandLine(getResourceCmd)
+	if err != nil {
+		klog.Errorf("Failed to get %v: %v", resType, err)
+		return []string{err.Error()}
+	}
+
+	names := []string{}
+	for _, o := range strings.Split(output, "\n") {
+		name := strings.TrimSpace(o)
+		if len(name) > 0 {
+			names = append(names, name)
+		}
+	}
+
+	return names
+}
+
+func GetMissionByName(name string) (*edgeclustersv1.Mission, error) {
+	getMissionCmd := fmt.Sprintf("%s get mission %s --kubeconfig=%s -o json ", config.Config.KubectlCli, name, config.Config.Kubeconfig)
+	output, err := util.ExecCommandLine(getMissionCmd)
+	if err != nil {
+		return nil, fmt.Errorf("Failed to get mission %v: %v", name, err)
+	}
+
+	var mission edgeclustersv1.Mission
+	err = json.Unmarshal([]byte(output), &mission)
+	if err != nil {
+		return nil, err
+	}
+
+	return &mission, nil
+}
+
+func TestClusterReady() bool {
+	testClusterCommand := fmt.Sprintf("%s cluster-info --kubeconfig=%s", config.Config.KubectlCli, config.Config.Kubeconfig)
+	if _, err := util.ExecCommandLine(testClusterCommand); err != nil {
+		klog.Errorf("The cluster is unreachable: %v", err)
+		return false
+	}
+
+	return true
+}
+
+// this is a hacky way for PoC only
```

Comment:
does this comment still apply? 

if yes, we need to create an issue to track this

Reply:
Created the tracking issue: https://github.com/CentaurusInfra/fornax/issues/28